src/uct/base/uct_md.h

```diff
@@ -14,15 +14,32 @@
 #include "uct_component.h"
 
 #include <uct/api/uct.h>
+#include <uct/api/v2/uct_v2.h>
 #include <ucs/config/parser.h>
 #include <ucs/memory/rcache.h>
+#include <ucs/type/param.h>
 #include <string.h>
 
 
 #define uct_md_log_mem_reg_error(_flags, _fmt, ...) \
     ucs_log(uct_md_reg_log_lvl(_flags), _fmt, ## __VA_ARGS__)
 
 
+#define UCT_MD_MEM_DEREG_FIELD_VALUE(_params, _name, _flag, _default) \
+    UCS_PARAM_VALUE(UCT_MD_MEM_DEREG, _params, _name, _flag, _default)
+
+
+#define UCT_MD_MEM_DEREG_CHECK_PARAMS(_params, _invalidate_supported) \
+    if (!UCT_MD_MEM_DEREG_FIELD_VALUE(_params, memh, FIELD_MEMH, NULL)) { \
+        return UCS_ERR_INVALID_PARAM; \
+    } \
+    if (!(_invalidate_supported) && \
+        (UCT_MD_MEM_DEREG_FIELD_VALUE(_params, flags, FIELD_FLAGS, 0) & \
```

Comment:
pls ad ENABLE_PARAM_CHECK to optimize-out the check in release mode

Reply:
added ENABLE_PARAMS_CHECK